src/libsyntax/json.rs

```diff
@@ -220,7 +222,7 @@ impl Diagnostic {
 
 impl DiagnosticSpan {
     fn from_span_label(span: SpanLabel,
-                       suggestion: Option<&String>,
+                       suggestion: Option<(&String, bool)>,
```

Comment:
I find it a bit weird that we have three different ways to represent the suggestion and whether it is machine applicable (two options, one option and a bool, one option of a tuple). It would be good to stick to one representation, if possible.